src/nginx/habitat/config/nginx.conf

```diff
@@ -0,0 +1,122 @@
+
+  sendfile on;
+  tcp_nopush on;
+  tcp_nodelay on;
+
+  keepalive_timeout 65;
+
+  gzip on;
+  gzip_http_version 1.0;
+  gzip_comp_level 2;
+  gzip_proxied any;
+  gzip_types text/plain text/css application/x-javascript text/xml application/xml application/xml+rss text/javascript application/json;
+
+  include {{pkg.svc_config_path}}/mime.types;
+
+  ## TODO Maybe put this back into a subdir
+  lua_package_path "{{pkg.svc_config_path}}/?.lua;$prefix/?.lua;;";
+  lua_shared_dict maint_data 1m;
+  lua_shared_dict banned_ips 1m;
+
+  # Preload some shared modules globally
+  init_by_lua '
+--    redis = require("resty/redis") # Not used in habitat chef server
+    config = require("config")
+    routes = require("routes")
+    resolver = require("resolver")
+    route_checks = require("route_checks")
+    validator = require("validator")
+
+-- Habitat doesnt support redis based dark launch and ban for now
+--   config.set_maint_refresh_interval(600)
+--   config.set_ban_refresh_interval(600)
+--   config.set_default_org(false)
+  ';
+
+  # Listen for local requests to the stub_status module.
+  server {
+    listen 127.0.0.1:9999;
+    location /nginx_status {
+      stub_status on;
+      access_log   off;
+      allow 127.0.0.1;
+      deny all;
+    }
+  }
+
+  upstream opscode_erchef {
+    {{~#if bind.oc_erchef}}
+    #  oc_erchef
+       {{~#eachAlive bind.oc_erchef.members as |member|}}
+         server {{ member.sys.ip}}:{{member.cfg.port}};
+       {{~/eachAlive}}
+    {{~/if}}
+  }
+  upstream oc_bifrost {
+    {{~#if bind.oc_bifrost}}
+    # oc_bifrost
+       {{~#eachAlive bind.oc_bifrost.members as |member|}}
+         server {{member.sys.ip}}:{{member.cfg.port}};
+       {{~/eachAlive ~}}
+    {{~/if}}
+  }
+  upstream opscode_solr4 {
+    {{~#if bind.elasticsearch}}
+       {{~#eachAlive bind.elasticsearch.members as |member|}}
+         server {{member.sys.ip}}:{{member.cfg.http-port}};
+       {{~/eachAlive}}
+    {{~/if}}
+  }
+  upstream bookshelf {
+    {{~#if bind.bookshelf}}
+       {{~#eachAlive bind.bookshelf.members as |member|}}
+         server {{member.sys.ip}}:{{member.cfg.port}};
+       {{~/eachAlive}}
+    {{~/if}}
+  }
+
+  # Include upstream definitions for addons
+  # TODO HABITAT plugins not yet
+  #  include /var/opt/opscode/nginx/etc/addon.d/*_upstreams.conf;
+
+  fastcgi_temp_path "{{pkg.svc_var_path}}/tmp/fastcgi";
+  client_body_temp_path "{{pkg.svc_var_path}}/tmp/client_body";
+  uwsgi_temp_path "{{pkg.svc_var_path}}/tmp/uwsgi";
+  scgi_temp_path "{{pkg.svc_var_path}}/tmp/scgi";
+
+  # external lb config for Chef API
+    proxy_cache_path  {{pkg.svc_data_path}}/cache/webui levels=1:2 keys_zone=webui-cache:50m max_size=5000m inactive=600m;
+    proxy_cache_path  {{pkg.svc_data_path}}/cache/cookbooks levels=1:2 keys_zone=cookbooks:50m max_size=5000m inactive=600m;
+    proxy_temp_path {{pkg.svc_data_path}}/cache-tmp;
+
+    # We support three options: serve nothing on non_ssl_port (80),
```

Comment:
We should sort out what set of the current configuration re ssl we want to support in this new setup.  I think we could make a strong argument for only including the SSL config.

Reply:
I lean towards only supporting non-ssl for sending redirects back to ssl. We could replace chef_http_lb.conf with a 301 block like the comment linked here.

Reply:
we need to keep http here, because customers will want to put an SSL-offloading loadbalancer in front of this.    However there's a neat trick we can do here.   newer nginx supports a syntax like this:

```
server {
  listen 80;
  listen 443 ssl http2 default_server;
  ...
}
```

we could then wrap the `listen 80;` line in some kind of conditional that can be disabled, and then debate what the default should be.   that also lets us get rid of a config file :)